frontend/amundsen_application/static/js/pages/TableDetailPage/ApplicationDropdown/index.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { Dropdown } from 'react-bootstrap';
+
+import AvatarLabel from 'components/AvatarLabel';
+import { TableApp } from 'interfaces';
+import { logClick } from 'utils/analytics';
+import AirflowMenu from './AirflowMenu';
+import DatabricksMenu from './DatabricksMenu';
+import GenericMenu from './GenericMenu';
+import {
+  APPLICATIONS_LABEL,
+  AIRFLOW,
+  DATABRICKS,
+  PRODUCING,
+  CONSUMING,
+} from './constants';
+import './styles.scss';
+
+export interface ApplicationDropdownProps {
+  tableApps: TableApp[];
+}
+
+const getImagePath = (tableAppName) => {
+  switch (tableAppName) {
+    case AIRFLOW.toLowerCase():
+      return '/static/images/airflow.jpeg';
```

Comment:
Nit: I would move this strings into constants, something like:
const AIRFLOW_LOGO_PATH
const DATABRICKS_LOGO_PATH
...